orca_client/src/send_bundle.rs

```diff
@@ -0,0 +1,127 @@
+use std::{str::FromStr, time::Duration};
+
+use futures::StreamExt;
+use jito_protos::{
+    bundle::Bundle,
+    convert::proto_packet_from_versioned_tx,
+    searcher::{
+        searcher_service_client::SearcherServiceClient, NextScheduledLeaderRequest,
+        SendBundleRequest, SubscribeBundleResultsRequest,
+    },
+};
+use jito_searcher_client::token_authenticator::ClientInterceptor;
+
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::{
+    commitment_config::CommitmentConfig,
+    pubkey::Pubkey,
+    signature::{read_keypair_file, Signature, Signer},
+    system_instruction::transfer,
+    transaction::{Transaction, VersionedTransaction},
+};
+use tokio::time::{sleep, timeout};
+use tonic::{codegen::InterceptedService, transport::Channel};
+
+type Client = SearcherServiceClient<InterceptedService<Channel, ClientInterceptor>>;
+
+/// Builds and send bundle, attaching the tip and waiting until the next leader
+pub async fn build_and_send_bundle(
+    rpc_url: String,
+    bundle: Vec<VersionedTransaction>,
+    keypair_path: String,
+    tip_lamports: u64,
+    tip_account: String,
+    client: &mut Client,
+) {
+    let payer_keypair = read_keypair_file(keypair_path).expect("reads keypair at path");
+    let tip_account = Pubkey::from_str(&tip_account).expect("valid pubkey for tip account");
+    let rpc_client = RpcClient::new(rpc_url);
+    let balance = rpc_client
+        .get_balance(&payer_keypair.pubkey())
+        .await
+        .expect("reads balance");
+
+    println!(
+        "payer public key: {:?} lamports: {:?}",
+        payer_keypair.pubkey(),
+        balance
+    );
+
+    let mut bundle_results_subscription = client
+        .subscribe_bundle_results(SubscribeBundleResultsRequest {})
+        .await
+        .expect("subscribe to bundle results")
+        .into_inner();
+
+    // wait for jito-solana leader slot
+    let mut is_leader_slot = false;
+    while !is_leader_slot {
+        let next_leader = client
+            .get_next_scheduled_leader(NextScheduledLeaderRequest {})
+            .await
+            .expect("gets next scheduled leader")
+            .into_inner();
+        let num_slots = next_leader.next_leader_slot - next_leader.current_slot;
+        is_leader_slot = num_slots <= 2;
+        println!("next jito leader slot in {} slots", num_slots);
+        sleep(Duration::from_millis(500)).await;
+    }
+
+    // build + sign the transactions
+    let blockhash = rpc_client
+        .get_latest_blockhash()
+        .await
+        .expect("get blockhash");
+
+    let tip_tx = VersionedTransaction::from(Transaction::new_signed_with_payer(
```

Comment:
run the risk of supplied txs being signed with an expired blockhash, how about accepting `Vec<Transaction>` instead of `Vec<VersionedTransaction>` as a param to this fn to ensure blockhash is fresh. fine to do later as well